src/main/webapp/WEB-INF/view/profile.jsp

```diff
@@ -22,38 +33,82 @@
+    <a href="/conversations">Conversations</a>
+    <% if (request.getSession().getAttribute("user") != null) { %>
+    	 <a href="/users/<%= request.getSession().getAttribute("user") %>" > <%= request.getSession().getAttribute("user") %>'s Profile</a>
+    <% } else { %>
+      	<a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
   </nav>
 
   <div id="container">
   	<div
 			style="width:75%; margin-left:auto; margin-right:auto; margin-top: 50px;">
-
-			<h1><strong><%=request.getSession().getAttribute("user")%>'s Profile Page</strong></h1>
-			<h2>Welcome!</h2>
 			
+			<%/** depending on who's page we are on we will see different name and content */ %>
+			<%
+			String sessionUser = (String) request.getSession().getAttribute("user");
+			String userProfile = (String) request.getAttribute("userProfile");
+			User currentUser = userStore.getUser(userProfile);
+			
+			if (userProfile.equals("")) { %>
+				<h1>Error This User Does Not Exist</h1>
+			
+			<% } else {
+					 if (sessionUser != null && sessionUser.equals(userProfile)){ %>
+						<h1 style="color:#3498DB"><strong>Welcome To Your Profile Page!</strong></h1>
+						<hr class="section-heading-spacer">
+					<% } else { %>
+						<h1><strong>Welcome to <%=userProfile %>'s Profile Page</strong></h1>
+						<hr class="section-heading-spacer">
+					<% } %>
+		
+			<%/** defult profile pic is a cute puppy */ %>
 			<div id="avatar">
-				<img alt="cute dog" src = "https://learnwebcode.com/images/lessons/insert-image-funny-dog.jpg">
+				<img alt="cute dog" src = "https://learnwebcode.com/images/lessons/insert-image-funny-dog.jpg" class="center">
 			</div>
-
+			<br/>
+			
+			
+			<%/** user's bio/aboutme section of profile */ %>
+			<% String profileBio = currentUser.getBiography(); %>
+			<h2 style="color:#083BF9">About <%=userProfile %> </h2>
+			<hr class="section-heading-spacer">
+			<a> <%= profileBio %></a>
+			<br/>
+			<br/>
+			
+			<%/** Edit profile bio aboutme */ %>
+			
+			<% if (sessionUser != null && sessionUser.equals(userProfile)) { %>
+				<a>Edit Your Bio Here <%=sessionUser %> (only you can see this)</a>
+				<form action="/users/<%=sessionUser %>" method="POST">
+					<input type="text" name="biography" value="<%= currentUser.getBiography() %>" >
+					<br/>
+				<button type="Submit">Submit</button>
+				</form>
+			<% } %>
+			
+			<hr class="section-heading-spacer">
 			<h3 class="font-semibold mgbt-xs-5"> Google CodeU Summer 2018 Student </h3>
-			<h4> University Student </h4>
+			<hr class="section-heading-spacer">
+			
+			<h2 style="color:indigo"> <%= userProfile %>'s Sent Messages </h2>
+			<hr class="section-heading-spacer">
+			<% List<Message> messagesSent = (List) request.getAttribute("messages");
+				for (Message message: messagesSent) { %>
+					<a><strong> <%=message.getTime() %> </strong> : <%= message.getContent() %></a>
+					<br/>
+				<% } %>
+				<hr class="section-heading-spacer">
+			<% } %>
```

Comment:
Is this the closing bracket for the if that checks if the userprofile exists?

Reply:
it checks if the session user is on their own profile, if so they can edit their bio